pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -679,7 +678,7 @@ cdef class StringHashTable(HashTable):
         for i in range(n):
             val = values[i]
 
-            if PyUnicode_Check(val) or PyString_Check(val):
```

Comment:
should note somewhere how we prefer isinstance checks for strings but use the cpython methods for datetimes.